src/main/java/org/minions/devfund/damianvp/battleship/Ocean.java

```diff
@@ -0,0 +1,196 @@
+            int placedShips = 0;
+            while (placedShips < shipQuantity) {
+                Ship ship = creator.getShip(shipType);
+                int row;
+                int column;
+                boolean horizontal;
+                do {
+                    row = random.nextInt(LENGTH - 1);
+                    column = random.nextInt(LENGTH - 1);
+                    horizontal = random.nextBoolean();
+
+                } while (!ship.okToPlaceShipAt(row, column, horizontal, this));
+                ship.placeShipAt(row, column, horizontal, this);
+                placedShips++;
+            }
+        }
+    }
+
+    /**
+     * method to verify if a specific position is occupied.
+     * @param row int value of specific position.
+     * @param column int value of specific position.
+     * @return true, if position is occupied.
+     *         false, if position is empty.
+     */
+    boolean isOccupied(int row, int column) {
+        return !ships[row][column].getShipType().equals("empty");
+    }
+
+    /**
+     * Method to shot at under specific position.
+     * @param row int value of specific position.
+     * @param column int value of specific position.
+     * @return true, if any ship was sunk in te position.
+     *         false, other case.
+     */
+    boolean shootAt(int row, int column) {
+        this.shotsFired++;
+        if (isOccupied(row, column)) {
+            if (ships[row][column].shootAt(row, column)) {
+                if (ships[row][column].isSunk()) {
+                    shipsSunk++;
+                }
+                hitCount++;
+                return true;
+            }
+            return false;
+        } else {
+            ships[row][column].shootAt(row, column);
+        }
+        return false;
+    }
+
+    /**
+     * Method to get the Ship board.
+     * @return the Ship board.
+     */
+    public Ship[][] getShips() {
+        return ships.clone();
+    }
+
+    /**
+     * Method to get the shot fired counter.
+     * @return the shotFired counter.
+     */
+    public int getShotsFired() {
+        return shotsFired;
+    }
+
+    /**
+     * Method to get the Hit counter.
+     * @return the Hit Counter.
+     */
+    public int getHitCount() {
+        return hitCount;
+    }
+
+    /**
+     * Method to get the ships sunk counter.
+     * @return ShipSunk counter.
+     */
+    public int getShipsSunk() {
+        return shipsSunk;
+    }
+
+    /**
+     * Method to verify if game is over.
+     * @return true if game is over.
+     *         false if game continue.
+     */
+    boolean isGameOver() {
+        return shipsSunk == TOTAL_SHIPS;
+    }
+
+    /**
+     * Method to get the Ship board.
+     * @return the Ship board.
+     */
+    Ship[][] getShipArray() {
+        return ships;
+    }
+
+    /**
+     * Method to print the ocean.
+     */
+    public void print() {
+        System.out.println(toString());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)